alert-database/src/main/java/com/synopsys/integration/alert/database/api/DefaultProviderDataAccessor.java

```diff
@@ -183,8 +194,28 @@ public void updateProjectAndUserData(final String providerName, final Map<Provid
     }
 
     private List<ProviderProject> updateProjectDB(final String providerName, final Set<ProviderProject> currentProjects) {
-        logger.info("{} projects", currentProjects.size());
-        return deleteAndSaveAllProjects(providerName, currentProjects);
+        final Set<ProviderProject> projectsToAdd = new HashSet<>();
+        final Set<ProviderProject> projectsToRemove = new HashSet<>();
+        final List<ProviderProject> storedProjects = findByProviderName(providerName);
+
+        projectsToRemove.addAll(storedProjects);
+        projectsToRemove.removeIf(project -> currentProjects
```

Comment:
We might want to create a HashSet of href to project to improve efficiency here since we're doing a lot of work inside of a transaction so cycles matter. We can then use `.containsKey(href)` on each element.

Reply:
If you had a HashSet<String> linkSet; the remove if would just change to removeIf(linkSet::contains)